app/index.ts

```diff
@@ -95,16 +95,16 @@ async function startAgent() {
         validateAgentConfig(config);
 
         logger.info('===============================================');
-        logger.info(`Initializing Saiki Agent in '${runMode}' mode...`, null, 'cyanBright');
+        logger.info(`Initializing Saiki in '${runMode}' mode...`, null, 'cyanBright');
         logger.info('===============================================\n');
 
         // Use the shared initializer
-        const { clientManager, llmService } = await initializeAgent(config, connectionMode);
+        const { clientManager, llmService } = await initializeServices(config, connectionMode);
 
         // Start based on mode
         if (runMode === 'cli' || runMode === 'both') {
             // Run CLI asynchronously, don't await if running both
-            const cliPromise = initializeAiCli(clientManager, llmService);
+            const cliPromise = await runAiCli(clientManager, llmService);
```

Comment:
i think we would be awaiting twice here in `cli` mode, not sure what happens when we do that.

and how would this work in `both` mode?